libs/expo/betterangels/src/lib/ui-components/AppUpdatePrompt/AppUpdatePrompt.tsx

```diff
@@ -0,0 +1,67 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import * as Updates from 'expo-updates';
+import { useEffect, useState } from 'react';
+import { useAppState } from '../../hooks';
+import { AppUpdatePromptModal } from './AppUpdatePromptModal';
+import { canShowPromptAgain } from './canShowPromptAgain';
+import { checkForUpdate } from './checkForUpdate';
+import { LAST_UPDATE_CHECK_TS_KEY, UPDATE_DISMISSED_TS_KEY } from './constants';
+
+export function AppUpdatePrompt() {
+  const [promptModalVisible, setPromptModalVisible] = useState(false);
+  const { appBecameActive } = useAppState();
+
+  useEffect(() => {
+    const showPrompt = async () => {
+      const canShowAgain = await canShowPromptAgain();
+
+      if (!canShowAgain) {
+        return;
+      }
+
+      const { isAvailable } = await checkForUpdate();
+
+      if (isAvailable) {
+        setPromptModalVisible(true);
+      }
+    };
+
+    if (!appBecameActive) {
+      return;
+    }
```

Comment:
Is this more appropriate logic?

```suggestion
if (!appBecameActive) return;

const canShowAgain = await canShowPromptAgain();
if (!canShowAgain) return;

const { isAvailable } = await checkForUpdate();
if (isAvailable) {
  setPromptModalVisible(true);
}

```

Reply:
can you compare it to this version the version in PR #1231 ?
I updated it there to allow bypassing the `canShowPromptAgain` for the Crash scenario